src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalProductOpt.java

```diff
@@ -64,20 +63,18 @@ public void getSupportedLocales(MessageCacheItem cacheItem) {
     public List<String> getComponents() {
         List<String> components = new ArrayList<String>();
         try {
-
             Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl());
-
-            URI uri = Thread.currentThread().getContextClassLoader().
-                    getResource(path.toString()).toURI();
-
-            if (uri.getScheme().equals("jar")) {
-                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
-                    path = fileSystem.getPath(path.toString());
+            List<URI> uris = FileUtil.findUris(path);
+            for (URI uri : uris) {
+                if (uri.getScheme().equals("jar")) {
+                    try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
+                        path = fileSystem.getPath(Paths.get(uri).toString());
+                        getComponents(path, components);
+                    }
+                } else {
+                    path = Paths.get(uri);
                     getComponents(path, components);
                 }
```

Comment:
Both if block and else block call 'getComponents(path, components);' , so it's better extract this line out of if-else block.

Reply:
Same reply as above. This was not changed in this PR.